### PR TITLE
feat: Add support for DEB triggers.

### PR DESCRIPTION
**Pull request by erikgeiser**
This pull request adds support for triggers in .deb files. These triggers are commonly used for tasks like rebuilding man pages or `initramfs`. Packages can activate a trigger for other packages or claim interest on triggers activated by other packages. For further information see https://wiki.debian.org/DpkgTriggers, https://man7.org/linux/man-pages/man5/deb-triggers.5.html.

In addition to the feature support, this pull request adds **unit tests**, an **acceptance test** and **documentation**. However, the following changes probably need some feedback before merging:
* The trigger names internally use a `-` (e.g. `activate-noawait`), however I used a `_` (e.g. `activate_noawait`)for the `nfpm` configuration to adhere to yaml conventions.
* `createControl` is now 2 or 3 lines above the line limit enforced by `golangci-lint:funlen`. I think it is still perfectly readable and simple enough, so I decided to ignore the warning by adding `// nolint:funlen`. I hope that's okay.
* In the acceptance test, the package only claims interest for one trigger which is then triggered manually. Activating a trigger is not explicitly tested. The main reason is that `accept()` currently does not support building two packages for the Docker container which would be necessary to test both ways of the trigger mechanism. However, the mechanism is simple enough that I think this test is sufficient, especially since the other cases are somewhat covered by the unit test.

### CODE REVIEW

File: www/docs/configuration.md [hunk 1]
```diff
@@ -125,6 +125,17 @@ deb:
...
     rules: foo.sh
+
+  # Custom deb triggers
+  triggers:
+    # register interrest on a trigger activated by another package
+    # (also available: interest-await, interest-noawait)
+    interest:
+      - some-trigger-name
+    # activate a trigger for another package
+    # (also available: activate-await, activate-noawait)
```
Comment (caarlos0) [inline www/docs/configuration.md:136]: if I understand correctly:

```suggestion
    # (also available: activate_await, activate_noawait)
```

File: www/docs/configuration.md [hunk 2]
```diff
@@ -125,6 +125,17 @@ deb:
   # Custom deb rules script.
   scripts:
     rules: foo.sh
+
+  # Custom deb triggers
+  triggers:
+    # register interrest on a trigger activated by another package
+    # (also available: interest-await, interest-noawait)
```
Comment (caarlos0) [inline www/docs/configuration.md:132]: I think it should be:

```suggestion
    # (also available: interest_await, interest_noawait)
```

**Comment by erikgeiser** [comment]
Yes, fixed it. I did a last minute change to adhere to the yaml conventions and I missed the docs.

**Comment by caarlos0** [comment]
can you resolve the conflicts as well? other than that I think we are good to merge :)

**Comment by erikgeiser** [comment]
Done.

**Comment by caarlos0** [comment]
thanks!

**Comment by caarlos0** [comment]
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**